internal/grpctest/tlogger.go

```diff
@@ -109,24 +109,47 @@ func Update(t *testing.T) {
 		grpclog.SetLoggerV2(logger)
 	}
 	logger.t = t
-	logger.errors = nil
+	logger.errors = map[*regexp.Regexp]int{}
 }
 
-// Expect declares an error to be expected. For the next test, all error logs
-// matching the expression (using FindString) will not cause the test to fail.
-// "For the next test" is includes all the time until the next call to Update().
-func Expect(expr string) {
+// ExpectError declares an error to be expected. For the next test, the first
+// error log matching the expression (using FindString) will not cause the test
+// to fail. "For the next test" is includes all the time until the next call to
+// Update(). Note that if an expected error is not encountered, this will cause
+// the test to fail.
+func ExpectError(expr string) {
+	ExpectErrorN(expr, 1)
+}
+
+// ExpectErrorN declares an error to be expected n times.
+func ExpectErrorN(expr string, n int) {
```

Comment:
I made them all methods of tLogger just because Tester is not instantiated for us to access its tLogger. (go test files do instantiate a struct embedding Tester and pass it to RunSubTests however)